pyro/contrib/funsor/infer/traceenum_elbo.py

```diff
@@ -39,9 +42,70 @@ def terms_from_trace(tr):
         # grab the log-density, found at all sites except those that are not replayed
         if node["is_observed"] or not node.get("replay_skipped", False):
             terms["log_factors"].append(node["funsor"]["log_prob"])
+    for p in terms["plate_vars"]:
+        terms["plate_to_step"][p] = terms["plate_to_step"].get(p, {})
     return terms
 
 
+@copy_docs_from(_OrigTraceEnum_ELBO)
+class TraceMarkovEnum_ELBO(ELBO):
```

Comment:
In a future PR, once we're confident that `TraceMarkovEnum_ELBO` is correct, we can just merge it with `TraceEnum_ELBO` instead of keeping the implementations separate.